# Exercise: Writing story pitch after analysis of a dataset.
## Dataset used: Periodic Labour Force Survey (PLFS), July 2026
## Published by: Ministry of Statistics and Programme Implementation, Govt. of India.

## Libraries used: pandas

## 1. Dataframe created using PLFS table data  for July 2026

In [10]:
import pandas as pd

data = {
    'Sector': ['Rural', 'Rural', 'Rural', 'Urban', 'Urban', 'Urban', 'Combined', 'Combined', 'Combined'],
    'Gender': ['Male', 'Female', 'Person', 'Male', 'Female', 'Person', 'Male', 'Female', 'Person'],
    'LFPR_July_2026': [78.3, 38.8, 58.0, 75.2, 25.3, 50.4, 77.2, 34.4, 55.4],
    'WPR_July_2026': [74.7, 37.2, 55.4, 70.7, 23.1, 47.0, 73.3, 32.5, 52.5],
    'UR_July_2026': [4.6, 4.3, 4.5, 5.9, 8.8, 6.7, 5.0, 5.4, 5.1]
}

# Create DataFrame
df_raw = pd.DataFrame(data)

# Save to CSV
file_name = "plfs_july_2026.csv"
df_raw.to_csv(file_name, index=False)

print(f" {file_name} has been created.")

 plfs_july_2026.csv has been created.


## 2. Basic characteristics of the PLFS table

In [4]:
import pandas as pd

df = pd.read_csv("plfs_july_2026.csv")

print("DF INFO")
df.info()

print("\nDF COLUMNS")
print(df.columns)

print("\nDF HEAD")
print(df.head(9))

DF INFO
<class 'pandas.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Sector          9 non-null      str    
 1   Gender          9 non-null      str    
 2   LFPR_July_2026  9 non-null      float64
 3   WPR_July_2026   9 non-null      float64
 4   UR_July_2026    9 non-null      float64
dtypes: float64(3), str(2)
memory usage: 492.0 bytes

DF COLUMNS
Index(['Sector', 'Gender', 'LFPR_July_2026', 'WPR_July_2026', 'UR_July_2026'], dtype='str')

DF HEAD
     Sector  Gender  LFPR_July_2026  WPR_July_2026  UR_July_2026
0     Rural    Male            78.3           74.7           4.6
1     Rural  Female            38.8           37.2           4.3
2     Rural  Person            58.0           55.4           4.5
3     Urban    Male            75.2           70.7           5.9
4     Urban  Female            25.3           23.1           8.8
5     Urban  Person            5

## 3. Analysis of women unemployment in the workforce

In [7]:
women_df = df[df['Gender'] == 'Female'].copy()

# Sorting: By Unemployment Rate (Highest to Lowest)
print("\n[WOMEN: UNEMPLOYMENT BY SECTOR]")
print(women_df.sort_values(by='UR_July_2026', ascending=False))


[WOMEN: UNEMPLOYMENT BY SECTOR]
     Sector  Gender  LFPR_July_2026  WPR_July_2026  UR_July_2026
4     Urban  Female            25.3           23.1           8.8
7  Combined  Female            34.4           32.5           5.4
1     Rural  Female            38.8           37.2           4.3


## 4. Comparing rural v/s urban female workforce participation

In [15]:
rural_val = women_df[women_df['Sector'] == 'Rural']['LFPR_July_2026'].values[0]
urban_val = women_df[women_df['Sector'] == 'Urban']['LFPR_July_2026'].values[0]
participation_gap = rural_val - urban_val

print(f"Rural Female LFPR: {rural_val}%")
print(f"Urban Female LFPR: {urban_val}%")
print(f"RESULTING PARTICIPATION GAP: {participation_gap:.2f}% (Rural leads Urban)")

Rural Female LFPR: 38.8%
Urban Female LFPR: 25.3%
RESULTING PARTICIPATION GAP: 13.50% (Rural leads Urban)


## 5. Calculating "Absorption ratio", i.e., percentage of willing participants who get employed.

In [14]:
women_df['Absorption_Index'] = (women_df['WPR_July_2026'] / women_df['UR_July_2026']).round(2)

print("\n[Absorption Index]")
print(women_df[['Sector', 'Absorption_Index']].sort_values(by='Absorption_Index', ascending=False))


[Absorption Index]
     Sector  Absorption_Index
1     Rural              8.65
7  Combined              6.02
4     Urban              2.62


# PITCH

## The Failure of Urban Workforce Absorption for Indian Women

The July 2026 PLFS data exposes a structural crisis. Indian cities currently provide an extremely inefficient environment for women to find work.

Key Facts:

1. Urban women participate far less in the workforce (25.3% vs. 38.8% in rural areas)
   The resultant participation gap is 13.5%
2. They remain unemployed at nearly double the rate (8.8% urban vs 4.3% rural).
3. The "Absorption Index" shows that rural labour markets are 3.3 times more efficient at turning a willing female worker into an employed worker than urban centers.

Until the urban unemployment rate for women (8.8%) is brought in line with the rural rate (4.3%) and educated, skilled women workers are included in the workforce, India's cities will continue to suffer rather than drive economic growth.